src/Filters/filt.jl

```diff
@@ -204,11 +216,13 @@ function filt!(out::AbstractVector, f::DF2TFilter{<:Biquad,<:Vector}, x::Abstrac
 end
 
 # Variant that allocates the output
-filt(f::DF2TFilter{<:FilterCoefficients{:z},<:Array{T}}, x::AbstractVector) where {T} =
-    filt!(Vector{T}(undef, length(x)), f, x)
+filt(f::DF2TFilter{<:FilterCoefficients{:z},<:Array{T}}, x::AbstractVector{V}) where {T,V} =
+    filt!(Vector{promote_type(T,V)}(undef, length(x)), f, x)
```

Comment:
Is there a reasonable use case for `promote_type(T,V) != T`? That would mean we cannot store the input samples in the state, at least not without loss of precision, right? Or am I missing something?

Reply:
Yes, but storing samples of a `Float64` signal as `Float32` is unusual to begin with. Whatever we do here, it may come as a surprise in one way or another, I guess. Would an identity filter convert the input to `Float32` and back to `Float64` or pass it through as is? Seems one would expect the latter. Is that true for the current implementations? But then a unit delay would suddenly cause loss of precision, which might also be a bit of a surprise.

So... I don't know. Your call unless anyone else wants to chime in.